setup/main.py

```diff
@@ -198,34 +198,76 @@ def __init_panel(self):
                                    Gio.SettingsBindFlags.GET)
 
         # custom theme
-        theme_name_list = self.__init_available_gtk_themes()
         self.__model_custom_theme = self.__builder.get_object(
                 "model_custom_theme")
-        for name in theme_name_list:
-            self.__model_custom_theme.append([name])
-        current_theme = self.__settings_panel.get_string('custom-theme')
-        try:
-            current_theme_number = theme_name_list.index(current_theme) + 1
-        except ValueError:
-            current_theme_number = 0
-
+        self.__combobox_custom_theme = self.__builder.get_object(
+                "combobox_custom_theme")
         self.__checkbutton_custom_theme = self.__builder.get_object(
                 "checkbutton_custom_theme")
+
+        def update_combobox_custom_theme(settings, key):
+            theme_name_list = self.__init_available_gtk_themes()
+            self.__model_custom_theme.clear()
+            for name in theme_name_list:
+                self.__model_custom_theme.append([name])
+            current_theme = self.__settings_panel.get_string(key)
+            try:
+                current_theme_number = theme_name_list.index(current_theme)
+            except ValueError:
+                self.__settings_panel.reset(key)
+                current_theme_number = theme_name_list.index(self.__settings_panel.get_string(key))
```

Comment:
Please separate the line.

current_icon = self.__settings_panel.get_string(key)
current_icon_number = icon_name_list.index(current_icon)